File: app/src/main/java/com/siva/virtual/util/DateConverter.java
package com.siva.virtual.util;
import androidx.room.TypeConverter;
import java.util.Date;

public class DateConverter {

    @TypeConverter
    public static Date getDateFromTimeStamp(Long value){
        return value == null ? null : new Date(value);
    }

    @TypeConverter
    public static Long getTimeFromDate(Date date){
        return date == null ? null : date.getTime();
    }
}

File: app/src/main/java/com/siva/virtualWallet/util/DateConverter.kt
package com.siva.virtualWallet.util

import androidx.room.TypeConverter
import java.util.*

object DateConverter {
    @JvmStatic
    @TypeConverter
    fun getDateFromTimeStamp(value: Long?): Date? {
        return if (value == null) null else Date(value)
    }

    @JvmStatic
    @TypeConverter
    fun getTimeFromDate(date: Date?): Long? {
        return date?.time
    }
}